src/api/routing.js

```diff
@@ -1,64 +1,37 @@
-import * as Http from '../http';
+import Http from '../http';
 import Navigator from '../navigator';
 
-// consts
-import STATUS from '../status_codes';
-
 class Routing {
-  static getDistanceToDestination(
-    {
+  static async getDistanceToDestination(routingOptions={}) {
+    if (!routingOptions.origin) {
+      const { latitude, longitude } = await Navigator.getCurrentPosition();
+      routingOptions.origin = { latitude, longitude };
+    }
+
+    let {
+      origin,
       destination,
       modes,
       units,
-    },
-    callback
-  ) {
-    Navigator.getCurrentPosition((status, { latitude, longitude }) => {
-      if (status !== STATUS.SUCCESS) {
-        callback(status);
-        return;
-      }
+    } = routingOptions;
 
-      this.getDistanceWithOrigin(
-        {
-          origin: {
-            latitude,
-            longitude,
-          },
-          destination,
-          modes,
-          units,
-        },
-        (status, routes) => {
-          callback(status, routes);
-          return;
-        },
-      );
-    });
-  }
+    origin = `${origin.latitude},${origin.longitude}`;
+
+    if (destination) {
+      destination = `${destination.latitude},${destination.longitude}`;
+    }
+    if (modes) {
+      modes = modes.join(',');
+    }
 
-  static getDistanceWithOrigin(
-    {
+    const params = {
```

Comment:
seems like we're constructing the `params` here with potentially `undefined` values, only to hydrate them out in `http.js` - alternatively, we could have `params = {}` at the top above `origin` and build up the map within the `if` conditions, that would avoid the extra filtering downstream

fine with either approach

Reply:
I like the filtering logic just being in one place (https) that applies to all functions calling it.